File: src/main/java/de/silencio/activecraftcore/guicreator/GuiBackItem.java
package de.silencio.activecraftcore.guicreator;

import de.silencio.activecraftcore.messages.GuiMessages;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.bukkit.ChatColor;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.inventory.meta.ItemMeta;

@Getter
@Setter
@EqualsAndHashCode(callSuper = false)
@ToString
public class GuiBackItem extends GuiItem {

    public GuiBackItem() {
        super(Material.ARROW);
        ItemMeta itemMeta = this.getItemMeta();
        setDisplayName(ChatColor.GOLD + GuiMessages.BACK_ARROW());
        this.setItemMeta(itemMeta);
        this.addClickListener(guiClickEvent -> {
            Player player = (Player) guiClickEvent.getView().getPlayer();
            if (GuiNavigator.getGuiStack(player) != null && GuiNavigator.getGuiStack(player).size() >= 1)
                GuiNavigator.pop(player);
        });
    }

}

File: src/main/java/org/activecraft/activecraftcore/guicreator/GuiBackItem.kt
package org.activecraft.activecraftcore.guicreator

import org.activecraft.activecraftcore.guicreator.GuiCreatorDefaults.backItemDisplayname
import org.bukkit.Material
import org.bukkit.entity.Player

class GuiBackItem constructor(
    displayName: String = backItemDisplayname(),
    material: Material = Material.ARROW
) : GuiItem(material) {

    init {
        setDisplayName(displayName)
        addClickListener { guiClickEvent ->
                val player = guiClickEvent.view.player as Player
                if (GuiNavigator.getGuiStack(player).size >= 1) GuiNavigator.pop(player)
        }
    }
}